clicommand/agent_pause.go

```diff
@@ -29,28 +29,18 @@ Example:
     $ buildkite-agent pause --note 'too many llamas' --timeout-in-minutes 60`
 
 type AgentPauseConfig struct {
+	GlobalConfig
+	APIConfig
+
 	Note             string `cli:"note"`
 	TimeoutInMinutes int    `cli:"timeout-in-minutes"`
-
-	// Global flags
-	Debug       bool     `cli:"debug"`
-	LogLevel    string   `cli:"log-level"`
-	NoColor     bool     `cli:"no-color"`
-	Experiments []string `cli:"experiment" normalize:"list"`
-	Profile     string   `cli:"profile"`
-
-	// API config
-	DebugHTTP        bool   `cli:"debug-http"`
-	AgentAccessToken string `cli:"agent-access-token" validate:"required"`
-	Endpoint         string `cli:"endpoint" validate:"required"`
-	NoHTTP2          bool   `cli:"no-http2"`
 }
 
 var AgentPauseCommand = cli.Command{
 	Name:        "pause",
 	Usage:       "Pause the agent",
 	Description: pauseDescription,
-	Flags: []cli.Flag{
+	Flags: append(globalAndAPIFlags(),
```

Comment:
This is a bit arbitrary and what you've done here works too, but another approach here could be to have the `globalAndAPIFlags()` function accept the flags as a variadic function?

```go
func withGlobalAndAPIFlags(additionalFlags ...[]cli.Flag) []cli.Flag {
  return append(globalAndAPIFlags(), additionalFlags)
}

// ... 
var AgentPauseCommand = cli.Command{
	Name:        "pause",
	Usage:       "Pause the agent",
	Description: pauseDescription,
	Flags: withGlobalAndAPIFlags(
    cli.StringFlag{
      // ...
    },
    cli.StringFlag{
      // ...
    },
    // ... etc
  ),

```

Reply:
What I was going for was expressing a command's flags as a composition of multiple groups of flags and its own specific flags, but the multiple `append`s was getting a bit ugly:

```go
append(append(globalFlags(), apiFlags()...), /* custom flags */ )
```

hence `globalAndAPIFlags`. A small helper function could replace `append` though, at the cost of wrapping the command-specific flags in a `[]cli.Flag{...}`, which doesn't seem all that bad actually.

Reply:
Nice. I like what you did with the `slices.Concat(globalFlags(), apiFlags(), []cli.Flag{ ...` 👏